Console customization doc page (#482)
kiali/kiali.io: content/en/docs/Configuration/console-customization.md

description: "Default selections, find and hide presets and custom metric aggregations."
---

## Custom metric aggregations

The inbound and outbound metric pages, in the _Metrics settings_ drop-down,
provides an opinionated set of groupings that work both for filtering out
metric data that does not match the selection and for aggregating data into
series. Each option is backed by a label on the collected Istio telemetry.

It is possible to add custom aggregations, like in the following example:

```yaml
spec:
  kiali_feature_flags:
    ui_defaults:
      metrics_inbound:
        aggregations:
        - display_name: Istio Network
          label: topology_istio_io_network
        - display_name: Istio Revision
          label: istio_io_rev
      metrics_outbound:
        aggregations:
        - display_name: Istio Revision
          label: istio_io_rev
```

Notice that custom aggregations for inbound and outbound metrics are defined separately.

You can find some [screenshots in Kiali v1.40 feature update blog
post](https://medium.com/kialiproject/kiali-release-1-40-features-update-78f19fd113c5#fe3d).

## Default metrics duration and refresh interval

> Most Kiali pages show a _metrics per refresh_ and a _refresh interval_

Reviewer: ```suggestion
Most Kiali pages show _metrics per refresh_ and _refresh interval_
```

drop-downs. These are located at the top-right of the page.

> The _metrics per refresh_ specifies the time range back from the current

Reviewer: ```suggestion
_Metrics per refresh_ specifies the time range back from the current
```

instant to fetch metrics and/or distributed tracing data. By default, a
1-minute time range is selected.

> The _refresh interval_ specifies how often Kiali will automatically refresh the

Reviewer: ```suggestion
_Refresh interval_ specifies how often Kiali will automatically refresh the
```

data shown. By default, Kiali refreshes data every 15 seconds.

```yaml
spec:
  kiali_feature_flags:
    ui_defaults:
      # Valid values: 1m, 5m, 10m, 30m, 1h, 3h, 6h, 12h, 1d, 7d, 30d
      metrics_per_refresh: "1m"

      # Valid values: pause, 10s, 15s, 30s, 1m, 5m, 15m
      refresh_interval: "15s"
```

User selections won't persist a reload.

## Default namespace selection

By default, when Kiali is accessed by the first time, on most Kiali pages users
will need to use the namespace drop-down to choose namespaces they want to view
data from. The selection will be persisted on reloads.

> However, it is possible to configure a set of predefined selection of

Reviewer: ```suggestion
However, it is possible to configure a predefined selection of
```

```

Namespace selection will reset to the predefined set on reloads. Also, if for
some reason a namespace becomes deleted, Kiali will simply ignore it from the
list.

## Graph find and hide presets

In the toolbar of the topology graph, the _Find_ and _Hide_ textboxes can be
configured with presets for your most used criteria. You can find [screenshots
and a brief description of this feature in the feature update blog post for
versions 1.31 to 1.33](https://medium.com/kialiproject/kiali-releases-1-34-to-1-39-overview-587f33fac41a#3962).

The following are the default presets:

```yaml
spec:
  kiali_feature_flags:
    ui_defaults:
      graph:
        find_options:
        - description: "Find: slow edges (> 1s)"
          expression: "rt > 1000"
        - description: "Find: unhealthy nodes"
          expression:  "! healthy"
        - description: "Find: unknown nodes"
          expression:  "name = unknown"
        hide_options:
        - description: "Hide: healthy nodes"
          expression: "healthy"
        - description: "Hide: unknown nodes"
          expression:  "name = unknown"
```

Hopefully, the attributes to configure this feature are self-explanatory.

> Note that by providing your own presets, you would be overriding the default

Reviewer: ```suggestion
Note that by providing your own presets, you will be overriding the default
```

configuration. Make sure to include any default presets that you need in case
you provide your own configuration.

> ## Graph default sources for traffic rates

Reviewer: ```suggestion
## Graph default traffic rates
```

Traffic rates in the graph are fetched from Istio telemetry and there are
several [metric sources](https://istio.io/latest/docs/reference/config/metrics/)
that can be used.

> In the graph page, you can select the rates source using the _Traffic_

Reviewer: ```suggestion
In the graph page, you can select the traffic rate metrics using the _Traffic_
```
